test/functional/deprecated_rpc.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test deprecation of RPC calls."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class DeprecatedRpcTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+
+    def setup_network(self):
+        self.nodes = []
+        self.nodes.append(self.start_node(0, self.options.tmpdir))
+        self.nodes.append(self.start_node(1, self.options.tmpdir,
+            ["-enablerpcmethod=estimatefee"]))
+
+    def run_test (self):
+        self.log.info("estimatefee: Shows deprecated message")
+        assert_raises_jsonrpc(-32, 'estimatefee is deprecated',
```

Comment:
no need for a lambda here, `assert_raises_jsonrpc()` should be called like this:

```python
assert_raises_jsonrpc(-32, 'estimatefee is deprecated', self.nodes[0].estimatefee, 1)
```